src/realm/group_shared.cpp

```diff
@@ -829,6 +829,7 @@ void SharedGroup::do_open(const std::string& path, bool no_create_file, bool is_
             File::UnmapGuard fug(m_file_map);
             SharedInfo* info_2 = m_file_map.get_addr();
             info_2->init_complete = 1;
+            m_file_map.sync();
```

Comment:
Flushing while holding an exclusive lock seems like something that could have a very bad effect on performance in some scenarios, because flushing is such an expensive operation.

I think we should only consider merging this, if we can prove that it is necessary.

However, if this turns out to be necessary, remember:

    bool disable_sync = get_disable_sync_to_disk();
    if (!disable_sync)
        m_file_map.sync(); // Throws